### PR TITLE
ci: move devnet test to make target

### PR DESCRIPTION
**Pull request by jyellick**
When encountering devnet test failures in CI, they're challenging to resolve and reproduce locally because the commands are embedded into the CI configuration and not the Makefile.

This PR simply extracts the commands from the CI steps and embeds them into the devnet python module, allowing for the python script to be invoked with a new `--test` flag to execute those steps.  It also adds a `devnet-test` make target to allow simple invocation of the script in test mode.

### CODE REVIEW

File: bedrock-devnet/devnet/__init__.py
Comment (tynes) [inline bedrock-devnet/devnet/__init__.py:86]: The idea here is that this test command is independent of bringing up the stack?

File: .circleci/config.yml
Comment (tynes) [inline .circleci/config.yml]: This doesn't look like it's running in CI, not sure if it's a permissions issue

**Review by tynes — COMMENTED** [review]
Generally in favor of the idea, not sure how I feel about the approach but could be open to it

File: bedrock-devnet/devnet/__init__.py
Comment (jyellick) [reply bedrock-devnet/devnet/__init__.py:86]: Correct.  To run the test manually, you'd do something like:

```
make devnet-up-deploy
make devnet-test
```

I'm not sure how everyone is using the `devnet` targets, but it seems plausible that some users are running other loads against the devnet, so I was wary of including the test execution with bringing the devnet up or down.

File: .circleci/config.yml
Comment (jyellick) [reply .circleci/config.yml]: I'm certainly not a circle CI expert, but it did execute in the source repo https://app.circleci.com/pipelines/github/bobanetwork/v3-anchorage/97/workflows/8f93a8dd-291f-4452-ad14-c47411905136/jobs/6089 so I believe the definition should be alright.

**Comment by jyellick** [comment]
> Generally in favor of the idea, not sure how I feel about the approach but could be open to it

I'm certainly happy to rework in some other way.  As best as I could tell, the assorted build scripts are being migrated to python (vs. bash), so I tried to keep everything in the python, even though it's simply executing commands.

If there's a different preferred approach, please let me know.  My goal here is primarily to allow developers to iterate while running tests without having to push it out and wait on CI.

**Comment by tynes** [comment]
> > Generally in favor of the idea, not sure how I feel about the approach but could be open to it
> 
> I'm certainly happy to rework in some other way. As best as I could tell, the assorted build scripts are being migrated to python (vs. bash), so I tried to keep everything in the python, even though it's simply executing commands.
> 
> If there's a different preferred approach, please let me know. My goal here is primarily to allow developers to iterate while running tests without having to push it out and wait on CI.

I am generally on board with this change after thinking about it more

**Review by ajsutton — COMMENTED** [review]
Love this change - moving stuff from `.circleci/config.yml` into makefiles is a long term goal.  Ideally we'd also move the `devnet (with genesis contracts)` tests as well, though that could be a separate PR as this is definitely an improvement.

If we do move the genesis tests, it would be great to remove the duplication as well so `make devnet-test` could just work for both genesis and deploy modes. I think the only difference is if `--l1-contracts-json-path` is specified or not so possibly it could switch based on whether that file exists or not.

File: bedrock-devnet/devnet/__init__.py
Comment (trianglesphere) [reply bedrock-devnet/devnet/__init__.py:86]: you would have `devnet-up-deploy` be a pre-requisite for `devnet-test` if you wanted `devnet-test` to automatically bring it up. The only potential issue is that it might always restart the devnet.

Comment (jyellick) [reply bedrock-devnet/devnet/__init__.py:86]: Right.  Conceivably, we could have multiple targets if we wanted to allow both, say `devnet-test-only` and `devnet-test` which required the `devnet-up-deploy` followed by `devnet-test-only`.  If there's some consensus, I'm happy to modify.

**Comment by jyellick** [comment]
I've updated the `--test` path of the python to add or omit the sdk address path depending on whether the file exists.  Now, things should work both for `make devnet-up` and `make devnet-up-deploy` -- I've also updated the circle CI definition to use it in both paths.

**Comment by jyellick** [comment]
Is there anything preventing this from merging?  Happy to make any desired changes if so.

**Comment by jyellick** [comment]
Rebased to try to keep this current (though no conflicts existed).  Any updates from the maintainers? @tynes @ajsutton ?

File: .circleci/config.yml
Comment (ajsutton) [inline .circleci/config.yml]: This piece doesn't seem to be in the python script so should probably stay here.

**Review by ajsutton — APPROVED** [review]
LGTM, other than I think we want to keep the capturing of op-node logs. I'm not an infra reviewer though.

**Comment by ajsutton** [comment]
Oh turns out I am an infra reviewer. :) @jyellick Would you mind restoring the capturing of op-node logs and then I'll get this merged.

Thank you so much for your incredible patience - we've been really slow on a bunch of your PRs.

File: .circleci/config.yml
Comment (jyellick) [reply .circleci/config.yml]: Oops! Good catch.  I hadn't originally removed this, but looks like it got clobbered in one of the rebases.  Fixed.

**Comment by ajsutton** [comment]
@jyellick The world is apparently out to get us today - the devnet build failed because of a network error downloading node. Can you kick it to rerun and then I think we should be good to go. It won't let me rerun because it look like its running in your circle ci org.

**Comment by jyellick** [comment]
Thanks @ajsutton, looks like the re-run fixed things, very happy to see this merge!

**Pull request merged**
